File: leetcode_problems_easy/1.two_sum/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {

	for i := 0; i < len(nums); i++ {

		for j := 0; j < len(nums); j++ {
			if nums[i]+nums[j] == target && j != i {
				sum := []int{i, j}
				return sum
			}
		}
	}
	return []int{0}
}

func main() {

	arr := []int{7, 2, 9}
	fmt.Println(twoSum(arr, 9))
}

File: leetcode_problems_easy/1.two_sum/two_sum.js
function twoSum(arr, target) {

	for(var i = 0; i < arr.length; i++) {
		
		for(var j = 1; j < arr.length; j++) {
			if (arr[i] + arr[j] === target && j !== i) {
				return [i, j]
			}
		}
	}
}


twoSum([3,7,2,15], 9)
twoSum([3,2,4], 6)
twoSum([3,3], 6)	
